### PR TITLE
Add form templates

### PR DESCRIPTION
**Pull request by josephsavona**
## Overview

Adds form templates for new issues:
- 🐛 Report a bug
- 🤦 Typo or mistake
- 💡 Suggestions
- 📄 Suggest new framework
- 📃 Bugs in React <link to react issues>
- 🤔 Questions and Help <link to support page>

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/0-bug.md
Comment (pwbriggs) [inline .github/ISSUE_TEMPLATE/0-bug.md:1]: @josephsavona if you're adding a new issue template, would it be worthwhile to write it as an [issue form](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) instead? What do you think?

Comment (rickhanlonii) [reply .github/ISSUE_TEMPLATE/0-bug.md:1]: Update: I converted this to a template and added all the form templates we need.

@pwbriggs, you spend a lot of time in the issues helping to triage, can you review the bug/typo templates and let me know what you think?

Comment (pwbriggs) [reply .github/ISSUE_TEMPLATE/0-bug.md:1]: (Haven't read through it all yet, will do that more thoroughly tomorrow.)

Looks like the link to open an issue in React is wrong. Also, I'd add `/choose` to the end to it goes to that repo's issue forms menu: `https://github.com/facebook/react/issues/new/choose`

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+contact_links:
+  - name: 📃 Bugs in React
+    url: https://github.com/facebook/react.dev/issues/new
+    about: This issue tracker is not for bugs in React. Please file React issues here.
```
Comment (harish-sethuraman) [inline .github/ISSUE_TEMPLATE/config.yml:4]: seems contradictory? 
```suggestion
    about: This issue tracker is not for bugs in React. Please file React docs issues here.
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+contact_links:
+  - name: 📃 Bugs in React
+    url: https://github.com/facebook/react.dev/issues/new
```
[the remark this answers is not in the document]
Comment (rickhanlonii) [reply .github/ISSUE_TEMPLATE/config.yml]: Nice catch, thanks

File: .github/ISSUE_TEMPLATE/0-bug.md
Comment (rickhanlonii) [reply .github/ISSUE_TEMPLATE/0-bug.md:1]: Fixed thanks

File: .github/ISSUE_TEMPLATE/config.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (rickhanlonii) [reply .github/ISSUE_TEMPLATE/config.yml]: This should actually go to facebook/react, I pushed the fix.

[the remark this answers is not in the document]
Comment (harish-sethuraman) [reply .github/ISSUE_TEMPLATE/config.yml]: my bad 😅

**Pull request merged**